File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        first=strs[0]
        last=strs[-1]
        i=0
        size=min(len(first),len(last))
        while i<size and first[i]==last[i]:
            i+=1
        if i==0:
            return ""
        return first[0:i:1]        

File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end());
        int i = 0, j = strs.size() - 1, idx = 0;
        string ans = "";
        if(i == j) return strs[i];
        while(idx < strs[i].size() and idx < strs[j].size() and strs[i][idx] == strs[j][idx]) ans += strs[i][idx], idx++;
        return ans;
    }
};